### PR TITLE
Add project query

### PR DESCRIPTION
**Pull request by Liliya-V**
created a php page to connect an add object form page to a database

### CODE REVIEW

File: adminPanel/addformDB.php [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
-$queryProjects= $db->prepare($sqlAdult);
+$queryProjects= $db->prepare($sqlProjects);
 $addProject = $queryProjects->execute([':title'=>$title, ':link'=>$link, ':image'=>$image]);
 
-echo 'Project was sucsessfully added';
\ No newline at end of file
+
+if (!$_POST['title'] || !$_POST['link'] || !$_POST['image']) {
+    echo  'Please, check if you filled all three lines';
+    echo'               .        ';
```
Comment (alicemcneil) [inline adminPanel/addformDB.php]: empty `echo'` here

File: adminPanel/addformDB.php [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
+    echo '<td><a href="addForm.php">Go back</a></td>';
+} else {
+    echo 'Project was sucsessfully added';}
```
Comment (alicemcneil) [inline adminPanel/addformDB.php]: *successfully

**Comment by alicemcneil** [comment]
Looks good! Just a few tiny things :) one of your commits is called 'q' which could be improved with a more descriptive commit name?

File: adminPanel/addForm.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+        <link href="../css/styleAdminPanel.css" rel="stylesheet" type="text/css">
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    </head>
+    <body>
+        <h1>Welcome to a project adding page, Liliya</h1>
+        <form  action="addformDB.php" method="POST">
+            <div class="container">
+               <label for="title">Type a title here:</label>
+               <input class="addform" type="text" name="title">
```
Comment (mporam) [inline adminPanel/addForm.php]: if you use a for attribute on the label tag it has to point at the id attribute on the input, so you need to add `id=title` to your input

Comment (mporam) [reply adminPanel/addForm.php]: and so on for the other inputs

File: adminPanel/addformDB.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
+<?php
+$db = new PDO("mysql:host=192.168.20.20;dbname=PortfolioProjects", 'root', '');
+$db -> setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+if(!empty($_POST['title'])) {
+    $title = filter_var($_POST['title'], FILTER_SANITIZE_STRING);
+    if(!empty($_POST['link'])) {
+        $site_url = filter_var($_POST['link'], FILTER_VALIDATE_URL);
+    }
+$title = $_POST['title'];
```
Comment (mporam) [inline adminPanel/addformDB.php]: uv already done this on line 5

File: adminPanel/addformDB.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+$link = $_POST['link'];
```
Comment (mporam) [inline adminPanel/addformDB.php]: uv got this in $site_url on line 7

File: adminPanel/addformDB.php [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+$image = 'images/' . $_POST['image'];
```
Comment (mporam) [inline adminPanel/addformDB.php]: no if statement to check you have an image?

File: adminPanel/addformDB.php (hunk 1 above)
Comment (mporam) [reply adminPanel/addformDB.php]: +1 why is this empty with a . ?

File: adminPanel/addformDB.php [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+$sqlProjects = "INSERT INTO `projects` (`title`, `link`, `image`) VALUES (:title, :link, :image);";
+$queryProjects= $db->prepare($sqlProjects);
+$addProject = $queryProjects->execute([':title'=>$title, ':link'=>$link, ':image'=>$image]);
+if (!$_POST['title'] || !$_POST['link'] || !$_POST['image']) {
+    echo  'Please, check if you filled all three lines';
+    echo'.';
+    echo '<td><a href="addForm.php">Go back</a></td>';
+} else {
+    echo 'Project was successfully added';}
```
Comment (mporam) [inline adminPanel/addformDB.php]: extra closeing `}`

**Review by mporam — CHANGES_REQUESTED** [review]

File: adminPanel/addformDB.php [hunk 8]
```diff
@@ -1,24 +1,24 @@
...
-} else {
-    echo 'Project was successfully added';}
+
+if(!empty($_POST['title'])) {
+    $title = filter_var($_POST['title'], FILTER_SANITIZE_STRING);
+    if (!empty($_POST['link'])) {
+        $link = filter_var($_POST['link'], FILTER_VALIDATE_URL);
+    }
 }
+if ($_POST['title'] || $_POST['link'] || $_POST['image']) {
```
Comment (mporam) [inline adminPanel/addformDB.php]: you want `&&` not `||`

Comment (mporam) [reply adminPanel/addformDB.php]: what happens if none of these exist? no error message?

Comment (mporam) [reply adminPanel/addformDB.php]: you should use `!empty()` on each of these

**Review by mporam — CHANGES_REQUESTED** [review]

File: adminPanel/addformDB.php (hunk 6 above)
Comment (Liliya-V) [reply adminPanel/addformDB.php]: done

File: adminPanel/addForm.php [hunk 9]
```diff
@@ -0,0 +1,28 @@
+<?php
+$db = new PDO("mysql:host=192.168.20.20;dbname=PortfolioProjects", 'root', '');
+$db -> setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
```
Comment (sjr-mayden) [inline adminPanel/addForm.php]: Spaces around the object accessor

File: adminPanel/addformDB.php [hunk 10]
```diff
@@ -0,0 +1,25 @@
+<?php
+require('functions.php');
+$db = new PDO("mysql:host=192.168.20.20;dbname=PortfolioProjects", 'root', '');
+$db -> setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
```
Comment (sjr-mayden) [inline adminPanel/addformDB.php]: Spaces around the object accessor

File: adminPanel/addformDB.php [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    $addProject = $queryProjects->execute(setVariables($_POST));
+    echo 'Project was successfully added.';
+    echo '<br><a href="addForm.php">Add another one. </a><br>';
+    echo '<a href="../portfolio.php">Go to your portfolio. </a>';
+} else {
+    echo 'Please fill all lines';
+    echo '<br><a href="addForm.php">Try again</a><br>';
+}
+
+
```
Comment (sjr-mayden) [inline adminPanel/addformDB.php]: Redundant trailing line breaks

File: adminPanel/functions.php [hunk 12]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+function validateProject($postData) {
```
Comment (sjr-mayden) [inline adminPanel/functions.php]: Missing type hinting for input and return * 2

File: adminPanel/test/functions.php [hunk 13]
```diff
@@ -0,0 +1,78 @@
...
+
+    public function testvalidateProject()
+    {
+        $postdata = ['title' => 'logo', 'link' => 'link', 'image' => 'image'];
+        $result = validateProject($postdata);
+        $this->assertInternalType('bool', $result);
+        $this->assertTrue($result);
+    }
+
+    public function testvalidateProject_failure()
```
Comment (sjr-mayden) [inline adminPanel/test/functions.php]: Why are these failing? That should be part of the name of the test function

**Review by sjr-mayden — CHANGES_REQUESTED** [review]
All changes requested are done, I have some minor pedantic things to do but nothing major

**Review by sjr-mayden — APPROVED** [review]

**Pull request merged**